scripts/commands/checkLinks.ts

```diff
@@ -94,7 +123,43 @@ async function determineFileBatch(args: Arguments): Promise<FileBatch> {
     toLoad.push("docs/api/qiskit/release-notes/index.md");
   }
 
-  return FileBatch.fromGlobs(toCheck, toLoad);
+  let description: string;
+  if (args.currentApis && args.qiskitReleaseNotes) {
+    description = "non-API docs, current API docs, and Qiskit release notes";
+  } else if (args.currentApis) {
+    description = "non-API docs and current API docs";
+  } else if (args.qiskitReleaseNotes) {
+    description = "non-API docs and Qiskit release notes";
+  } else {
+    description = "non-API docs";
+  }
+
+  return FileBatch.fromGlobs(toCheck, toLoad, description);
+}
+
+async function determineHistoricalFileBatches(
+  projectName: string,
+): Promise<FileBatch[]> {
+  const versionIndexPaths = await globby(
+    `docs/api/${projectName}/[0-9]*/index.md`,
+  );
+  versionIndexPaths.sort();
+
+  const result = [];
+  for (const indexPath of versionIndexPaths) {
+    const versionMatch = indexPath.match(/(\d+\.\d+)/);
+    if (versionMatch === null) {
+      throw new Error(`Failed to determine version from ${indexPath}`);
+    }
+
+    const fileBatch = await FileBatch.fromGlobs(
+      [indexPath.replace("index.md", "*")],
+      [],
+      `${projectName} v${versionMatch[0]}`,
+    );
+    result.push(fileBatch);
+  }
+  return result;
 }
```

Comment:
FYI, alternative method: To create all the paths for the historical versions, maybe we can read directly the names of the folders we have. The advantage is that they are already sorted by name, and we don't need to use the replace for the glob expression. Notice that we need to import `readdir` as well: `import { readdir } from "fs/promises";
`
```suggestion
async function determineHistoricalFileBatches(
  projectName: string,
): Promise<FileBatch[]> {
  const historicalFolders = (
    await readdir(`docs/api/${projectName}`, { withFileTypes: true })
  ).filter((file) => file.isDirectory() && file.name.match(/[0-9].*/));

  const result = [];
  for (const folder of historicalFolders) {
    const fileBatch = await FileBatch.fromGlobs(
      [`docs/api/${projectName}/${folder.name}/*`],
      [],
      `${projectName} v${folder.name}`,
    );
    result.push(fileBatch);
  }
  return result;
}
```

Reply:
Much better! Thanks!